fvm/src/machine/engine.rs

```diff
@@ -83,25 +129,30 @@ impl Deref for Engine {
 }
 
 impl Engine {
-    /// Create a new Engine from a wasmtime config.
-    pub fn new(c: &wasmtime::Config) -> anyhow::Result<Self> {
-        Ok(wasmtime::Engine::new(c)?.into())
+    pub fn new_default(ec: EngineConfig) -> anyhow::Result<Self> {
+        Engine::new(&default_wasmtime_config(), ec)
     }
-}
 
-impl From<wasmtime::Engine> for Engine {
-    fn from(engine: wasmtime::Engine) -> Self {
-        Engine(Arc::new(EngineInner {
+    /// Create a new Engine from a wasmtime config.
+    pub fn new(c: &wasmtime::Config, ec: EngineConfig) -> anyhow::Result<Self> {
+        let engine = wasmtime::Engine::new(c)?;
+
+        let mut dummy_store = wasmtime::Store::new(&engine, ());
+        let gg_type = GlobalType::new(ValType::I64, Mutability::Var);
+        let dummy_gg = Global::new(&mut dummy_store, gg_type, Val::I64(0))
+            .expect("failed to create dummy gas global");
+
```

Comment:
Now that we're doing even more at engine init, we should really be pooling these engines.

Reply:
We pool them in the FFI (using the `MultiEngine` as a "pool").

Reply:
That's what the MultiEngine is doing, and we were already re-using the engine in filecoin-ffi.